### DISCUSSION TITLE
Npm package badges

### DISCUSSION THREAD

**Discussion by Judimax** [community/community#134539] [npm]
Greetings CircleCI has repo badges to give various metadata such as CI/CD testing and the sort does npm have something similar?

**Comment by oguzhan18** [comment]
Hey @Judimax,

That's a solid question, and I see where you're coming from. While npm itself doesn't provide built-in badges like CircleCI for CI/CD metadata, there are definitely some well-established tools you can leverage to get the job done.

Here are a few services you should check out:

### 1. **Shields.io**
This is one of the go-to services for generating various badges. You can create badges for things like package version, download stats, and even license information. A couple of examples for your npm package would be:

- Version: 

 
`  ![npm version](https://img.shields.io/npm/v/your-package-name.svg)`
 

- Downloads: 

 
     `![npm downloads](https://img.shields.io/npm/dm/your-package-name.svg)`
 
### 2. ***Badgen.net***
If you're looking for something that's fast and lightweight, Badgen.net is another good alternative to Shields.io. It also supports npm-specific metadata, along with badges for other services like GitHub actions.

### 3. ***CI Badges***
For build statuses, even though npm doesn’t handle this directly, you can still integrate CI/CD badges into your README if you're using services like GitHub Actions, Travis CI, or CircleCI. It gives your repo a nice touch by displaying the current state of your builds (passing/failing).

### 4. ***Snyk***
If you’re concerned about the security of your package, you can include a Snyk badge to showcase the security status of your npm dependencies. It’s a great way to signal that you're keeping things secure.

**Comment by dinosy817** [comment]
Yes, npm provides a way to display badges that indicate the status of your package, similar to CircleCI badges. You can add badges to your project’s README to showcase important metadata. Here are some common badges you might want to include:
1. npm Version Badge
To show the latest version of your package on npm:
`![npm](https://img.shields.io/npm/v/your-package-name.svg)`
2. npm Downloads Badge
To display the number of weekly downloads:
`![npm](https://img.shields.io/npm/dt/your-package-name.svg)`
3. License Badge
To indicate the license of your package:
`![License](https://img.shields.io/npm/l/your-package-name.svg)`
4. CI Build Status Badge
If you're using a CI/CD service, you can include its build status badge. For example, if you're using CircleCI, you would use:
`[![CircleCI](https://circleci.com/gh/your-username/your-repo.svg?style=svg)](https://circleci.com/gh/your-username/your-repo)`
